applicationset/webhook/webhook.go

```diff
@@ -178,7 +180,11 @@ func (h *WebhookHandler) Handler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	h.HandleEvent(payload)
+	h.Add(1)
+	go func() {
+		defer h.Done()
+		h.HandleEvent(payload)
+	}()
```

Comment:
This is a problem with existing implementation as well. This changes simply terminates the go routine handling the request and creates a new one, instead of continuing the request go routine. IMO, this change is no worse at handling the spam than existing implementation (since go can create thousands of go routines without breaking a sweat). We are more likely to get OOM in case of spam, since each request independently pulls all Apps/AppSet in memory than noticing any problems with go routines.

Reply:
(Thanks for thoughtful comments, could address it as I have been busy for some time)
I have updated the code to use the worker pool model with configurable concurrency.